redash/query_runner/mongodb.py

```diff
@@ -93,39 +93,34 @@ def _get_column_by_name(columns, column_name):
     return None
 
 
+def _parse_dict(dic):
+    res = {}
+    for key, value in dic.items():
+        if isinstance(value, dict):
+            for tmp_key, tmp_value in _parse_dict(value).items():
+                new_key = "{}.{}".format(key, tmp_key)
+                res[new_key] = tmp_value
+        else:
+            res[key] = value
+    return res
+
+
 def parse_results(results):
     rows = []
     columns = []
 
     for row in results:
         parsed_row = {}
 
-        for key in row:
-            if isinstance(row[key], dict):
-                for inner_key in row[key]:
-                    column_name = "{}.{}".format(key, inner_key)
-                    if _get_column_by_name(columns, column_name) is None:
```

Comment:
I think you can remove `_get_column_by_name` now

Reply:
I will try.

Reply:
It has been used in unit test cases, I will keep it currently.

Reply:
It seems it's still used in this file in the new line 339, old line 344.